# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerParent.h
# Commit: 9e22fbf8ab8f
# Full Hash: 9e22fbf8ab8f828b6e500a6e4994accc42c5e819
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r a5880ebdc3d7 -r 9e22fbf8ab8f dom/workers/remoteworkers/RemoteWorkerDebuggerParent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerParent.h	Tue Apr 08 09:43:33 2025 +0000
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_RemoteWorkerDebuggerParent_h
+#define mozilla_dom_RemoteWorkerDebuggerParent_h
+
+#include "mozilla/dom/PRemoteWorkerDebuggerParent.h"
+#include "mozilla/dom/PRemoteWorkerDebuggerChild.h"
+#include "mozilla/ipc/Endpoint.h"
+#include "nsIWorkerDebugger.h"
+#include "nsTArray.h"
+
+namespace mozilla::dom {
+
+class RemoteWorkerDebuggerParent final : public PRemoteWorkerDebuggerParent,
+                                         public nsIWorkerDebugger {
+  friend class PRemoteWorkerDebuggerParent;
+
+  NS_DECL_THREADSAFE_ISUPPORTS
+  NS_DECL_NSIWORKERDEBUGGER
+
+ public:
+  RemoteWorkerDebuggerParent(
+      const RemoteWorkerDebuggerInfo& aWorkerDebuggerInfo,
+      Endpoint<PRemoteWorkerDebuggerParent>&& aParentEp);
+
+  mozilla::ipc::IPCResult RecvUnregister();
+
+  mozilla::ipc::IPCResult RecvReportErrorToDebugger(
+      const RemoteWorkerDebuggerErrorInfo& aErrorInfo);
+
+  mozilla::ipc::IPCResult RecvPostMessageToDebugger(const nsString& aMessage);
+
+  mozilla::ipc::IPCResult RecvSetAsInitialized();
+  mozilla::ipc::IPCResult RecvSetAsClosed();
+
+  mozilla::ipc::IPCResult RecvAddWindowID(const uint64_t& aWindowID);
+  mozilla::ipc::IPCResult RecvRemoveWindowID(const uint64_t& aWindowID);
+
+ private:
+  ~RemoteWorkerDebuggerParent();
+
+  bool mIsInitialized{false};
+  bool mIsClosed{false};
+  RemoteWorkerDebuggerInfo mWorkerDebuggerInfo;
+
+  nsTArray<uint64_t> mWindowIDs;
+
+  nsTArray<nsCOMPtr<nsIWorkerDebuggerListener>> mListeners;
+};
+
+}  // namespace mozilla::dom
+
+#endif  // mozilla_dom_RemoteWorkerDebuggerParent_h